src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -782,10 +783,33 @@ public void run() {
                 return PlaceholderTask.this;
             }
 
+            @Exported
+            public Integer getNumber() {
+                Run<?, ?> r = getParent().runForDisplay();
+                return r != null ? r.getNumber() : -1;
```

Comment:
I think it was fine (probably preferable) to return `null` here as you had it previously - it just looked like the test needed to be fixed to handle `null` correctly.